warehouse/rate_limiting/__init__.py

```diff
@@ -162,7 +163,82 @@ def __eq__(self, other):
         )
 
 
+def ratelimit_view_deriver(view, info):
+    """
+    A general-purpose rate limit view deriver based on client's `remote_addr`.
+    https://docs.pylonsproject.org/projects/pyramid/en/latest/narr/hooks.html#custom-view-derivers
+
+    Add `ratelimiter="<limit-string>"` to a `@view_config()`. Example:
+
+    ```
+    @view_config(
+        ...,
+        ratelimit="10/second",
+        ratelimit_by="user.id",  # Optional, defaults to "remote_addr"
+    )
+    ```
+    https://limits.readthedocs.io/en/stable/quickstart.html#rate-limit-string-notation
+    """
+    limit_str = info.options.get("ratelimit")
+    limit_by = info.options.get("ratelimit_by", "remote_addr")
+    if limit_str:
+
+        def wrapper_view(context, request):
+            limit_attr = getattr(request, limit_by)
+
+            ratelimiter = request.find_service(
+                IRateLimiter, name="rate_limiting.client"
+            )
+            # TODO: Is this Kosher?
+            # Override the default limits with the ones specified in the view
+            ratelimiter._limits = parse_many(limit_str)
+
+            metrics = request.find_service(IMetricsService, context=None)
+
+            request_route = request.matched_route.name
+
+            ratelimiter.hit(limit_attr)
```

Comment:
What's pretty odd is that this works but doesn't appear to add entries to redis as far as I can tell. The `ratelimiter._storage` is a `RedisStorage`, and I can get rate limited just fine, but I never see a key inserted into redis.